File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        string s ="";
        sort(strs.begin(), strs.end());
        string first = strs[0], last = strs[strs.size() - 1];
        for(int i = 0; i < first.size(); i++){
            if(first[i] == last[i]) s += first[i];
            else break;
        }
        return s;
    }

};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        s = ""
        shortest = min(strs, key = len)
        for i in range(len(shortest)):
            for ch in strs :
                if shortest[i] != ch[i]:
                    return s
            else:
                s += shortest[i]  
        return s                     